### ISSUE TITLE
Cross-compiling for darwin fails with "library not found for -lcrt0.o"

### ISSUE THREAD

**Issue by kolaente** [techknowlogick/xgo#79]
When cross-compiling for darwin with the newest version, building fails with these errors:

```
/usr/local/go/pkg/tool/linux_amd64/link: running o64-clang failed: exit status 1
ld: library not found for -lcrt0.o
clang: error: linker command failed with exit code 1 (use -v to see invocation)

2020/09/03 18:31:18 Failed to cross compile package: exit status 2.
Error: exit status 1
```

[Full error message in CI](https://drone.kolaente.de/vikunja/api/2153/2/7).

The code is [here](https://code.vikunja.io/api).

This worked until recently so I'm not sure if this is a bug in my code or with xgo.

**Comment by kolaente** [comment]
There seem to be various (rather old) issues around this, also ones related to go: 

* https://github.com/golang/go/issues/6718 
* http://www.cplusplus.com/forum/beginner/90152/
* https://groups.google.com/forum/#!topic/golang-dev/JKM7UaKRT6g

The error suggests it's an issue with missing a missing library but since it worked before I'm not sure if thats the cause.

**Comment by kolaente** [comment]
So far, I've tried a few things, including different go versions and checking out a working git commit of my project. None of it worked.

I then realized I messed up the xgo invocation command for darwin

The command I ran which did not work was this:
```
xgo -dest ./dist/binaries -tags netgo  -ldflags '-linkmode external -extldflags "-static" -X "code.vikunja.io/api/pkg/version.Version=v0.14.0+72-78b261e440" -X "main.Tags="' -targets "darwin/*"  -out vikunja-master .
```

Once I removed the `-linkmode external -extldflags "-static" ` part from it, resulting in a command like this:
```
xgo -dest ./dist/binaries -tags netgo  -ldflags '-X "code.vikunja.io/api/pkg/version.Version=v0.14.0+72-78b261e440" -X "main.Tags="' -targets "darwin/*"  -out vikunja-master .
```

It worked!

I'm not 100% sure as to why that worked and the previous one didn't, but [this help article](https://developer.apple.com/library/archive/qa/qa1118/_index.html) seems to indicate it is not possible to create binaries for darwin with statically linked dependencies. There seems to be [hacks](https://dropline.net/2015/10/static-linking-on-mac-os-x/) available to make this work - maybe one day this could be incorporated into xgo.
